File: ClassDataArray/ClassDataApp.py
class Person:
    def __init__(self,firstname,lastname,age):
        self.firstname=firstname
        self.lastname=lastname
        self.age=age
    
class ClassDataArray:
    person_list=[]
    
    def insert(self,personObj):
        self.person_list.append(personObj)
        return "sucessfully inserted"
    
    def find(self,firstname):
        if len(self.person_list)==0:
            return "list is empty"
        for i in self.person_list:
            if i.firstname==firstname:
                return i.firstname+" "+i.lastname+" "+str(i.age)
        return "person not found"
    
    def delete(self,firstname):
        if len(self.person_list)==0:
            return "list is empty"
        plist=self.person_list
        for i in range(len(plist)):
            if plist[i].firstname==firstname:
                plist.pop(i)
                self.person_list=plist
                return "succesfully deleted"
        return "person not found"
    
    def displayAll(self):
        if len(self.person_list)==0:
            return "list is empty"
        for i in self.person_list:
            print(i.firstname+" "+i.lastname+" "+str(i.age))
        return "sucessfully printed entire list"
class ClassDataApp:
    def main():
        arr=ClassDataArray()
        while(True):
            print("1.insert  2.delete  3.printall  4.find  5.exit")
            choice=int(input("enter your choice: "))
            if(choice==1):
                pobj=Person(input("enter firstname: "),input("enter Lastname: "),int(input("enter age: ")))
                print(arr.insert(pobj))
            elif(choice==2):
                print(arr.delete(input("enter firstname to delete: ")))
            elif(choice==3):
                print(arr.displayAll())
            elif(choice==4):
                print(arr.find(input("enter firstname to find: ")))
            else:
                break   
    if __name__=="__main__":
        main()

File: ClassDataArray/ClassDataApp.cpp
#include <bits/stdc++.h>
using namespace std;

class Person
{
  public:
  string FirstName;
  string LastName;
  int Age;
  Person()
  {
       
  }
  Person(string fname,string lname,int age)
  {
    FirstName=fname;
    LastName=lname;
    Age=age;
  }


};

class ClassDataArray
{
  public:
  int size=0,i,j;

  ClassDataArray()
  {

  }

  int insert(string fname,string lname,int age,Person arr[])
  {
     Person pobj=Person(fname,lname,age);
     arr[size]=pobj;
     size++;
     cout<<"Succesfully inserted new record with firstname "<<fname<<endl;
     return size;
  }

  void find(string fname,Person arr[])
  {
     for(i=0;i<size;i++)
    {
      if(fname==arr[i].FirstName)
      {
         cout<<"Record Found :"<<endl;
         cout<<arr[i].FirstName<<" "<<arr[i].LastName<<" "<<arr[i].Age<<endl;
         break;
      }
    }

  }
  int Delete(string fname,Person arr[])
  {
       for(i=0;i<size;i++)
    {
      if(fname==arr[i].FirstName)
      {
         for(j=i;j<size-1;j++)
         {
           arr[j]=arr[j+1];
         }
         break;
      }
    }
    size--;
    cout<<"succesfully deleted "<<fname<<" from the records"<<endl;
    return size;
  }

 
  void printAll(Person arr[])
  {
    cout<<"Printing All Records: "<<endl;
    for(i=0;i<size;i++)
    {
      cout<<arr[i].FirstName<<" "<<arr[i].LastName<<" "<<arr[i].Age<<endl;
      cout<<endl;
    }
  }
};

int main()
{
  int cursize=0,maxsize,choice,age;
     string fname,lname;
     cout<<"enter the size of array: ";
     cin>>maxsize;
     Person arr[maxsize];
     ClassDataArray array =ClassDataArray();
     while(1)
     {
         cout<<"1.Insert  2.Find  3.Delete  4.Printall 5.Exit"<<endl;
         cout<<"enter your choice: ";
         cin>>choice;
         
         if(choice==1)
         {
           if(cursize>=maxsize)
           {
             cout<<"Array is full"<<endl;
             continue;
           }
            cout<<"enter firstname: ";
            cin>>fname;
            cout<<"enter lastname: ";
            cin>>lname;
            cout<<"enter age: ";
            cin>>age;
            cursize=array.insert(fname,lname,age,arr);
         }
         else if(choice==2)
         {
           if(cursize==0)
           {
             cout<<"Array is Empty"<<endl;
             continue;
           }
             cout<<"enter firstname to find: ";
             cin>>fname;
             array.find(fname,arr);
         }
         else if(choice==3)
         {
           if(cursize==0)
           {
             cout<<"Array is Empty"<<endl;
             continue;
           }
             cout<<"enter firstname to delete: ";
             cin>>fname;
             cursize=array.Delete(fname,arr);
         }
         else if(choice==4)
         {
           if(cursize==0)
           {
             cout<<"Array is Empty"<<endl;
             continue;
           }
            array.printAll(arr);
         }
         else
         {
           break;
         }
     }
      return 0;
}
